src/base1/cockpit.js

```diff
@@ -4286,38 +4298,31 @@ function factory() {
             if (user.id === 0)
                 return true;
 
-            if (user.groups) {
-                var allowed = false;
-                user.groups.forEach(function(name) {
-                    if (name == group) {
-                        allowed = true;
-                        return false;
-                    }
-                    if (admin && (name == "wheel" || name == "sudo")) {
-                        allowed = true;
-                        return false;
-                    }
-                });
-                return allowed;
-            }
+            if (group)
+                return !!(user.groups || []).includes(group);
+
+            if (admin)
+                return self.is_superuser;
 
             if (user.id === undefined)
                 return null;
 
             return false;
         }
 
-        if (self.user) {
+        if (self.user && self.is_superuser !== null) {
             self.allowed = decide(self.user);
         } else {
-            cockpit.user().done(function (user) {
-                self.user = user;
-                var allowed = decide(user);
-                if (self.allowed !== allowed) {
-                    self.allowed = allowed;
-                    self.dispatchEvent("changed");
-                }
-            });
+            cockpit.all([cockpit.user(), check_superuser()])
+                .done(function (user, is_superuser) {
```

Comment:
@larskarlitski : This might interfere with your fixing of `cockpit.all`. Should this be written differently?

Reply:
I haven’t yet figured out a completely backwards compatible way to fix the `cockpit.all` patch. It would probably be best if you remove the array in the argument list and use varargs, because you also expect varargs in the function you pass to `done`. But I can also fix it as part of that other patch, as soon as I come up with a solution. Thanks for thinking about it!

Reply:
Ack, I moved to the more consistent varargs form.